src/session.ts

```diff
@@ -30,66 +29,37 @@ export interface SessionContext<S extends object> extends Context {
  * or pass custom `storage`.
  *
  * @example https://github.com/telegraf/telegraf/blob/develop/docs/examples/session-bot.ts
- * @deprecated https://github.com/telegraf/telegraf/issues/1372#issuecomment-782668499
  */
-export function session<S extends object>(
-  options?: SessionOptions<S>
-): MiddlewareFn<SessionContext<S>> {
-  const getSessionKey = options?.getSessionKey ?? defaultGetSessionKey
-  const store = options?.store ?? new MemorySessionStore()
+export function session<S extends object>({
+  getSessionKey = defaultGetSessionKey,
+  store = new Map(),
+}: SessionOptions<S> = {}): MiddlewareFn<SessionContext<S>> {
   return async (ctx, next) => {
-    const key = await getSessionKey(ctx)
+    const key = getSessionKey(ctx)
     if (key == null) {
       return await next()
     }
-    ctx.session = await store.get(key)
+    Object.defineProperty(ctx, 'session', {
+      get() {
+        return store.get(key)
+      },
+      set(value) {
+        if (value === undefined) {
```

Comment:
Maybe check for nullish?

Reply:
I.e., handle the case where users assign `null`. We way want to handle this the same way as for `undefined`.

Reply:
```suggestion
        if (value == null) {
```

Reply:
I understand, but I don't see any benefit of (effectively) replacing `null` with `undefined`.

Reply:
The point is that telegraf does this in all places, it checks for nullish instead of strictly undefined. So why suddenly do it differently here?

Reply:
We don't iterate over keys, so from JS PoV, the `.delete(key)` is just memory-efficient `.set(key, undefined)`, not `.set(key, null)`.

Reply:
If someone ignores the types and sets `ctx.session` to `null`, why should `ctx.session` be `undefined` and not `null`?

Reply:
Could you link the previous example?

Reply:
b1c0e3c238a33ce6a367e8efd5ce2349c5eeba5f

Reply:
But I think I was thinking about this: https://github.com/telegraf/telegraf-session-firebase#destroying-a-session